From 78a35482b1e7d5a8d75fd015010ab7ced32eb33d Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:29:53 -0400
Subject: [net] net/openvswitch: Use with RCU_INIT_POINTER(x, NULL) in vport-gre.c

Message-id: <c818232383c49b79bb0f63344783d13b38aaa7b1.1410531783.git.jbenc@redhat.com>
Patchwork-id: 94025
O-Subject: [RHEL7.1 net PATCH 20/50] net/openvswitch: Use with RCU_INIT_POINTER(x, NULL) in vport-gre.c
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 944df8ae84d88f5e8eb027990dad2cfa4fbe4be5
Author: Monam Agarwal <monamagarwal123@gmail.com>
Date:   Mon Mar 24 00:52:43 2014 +0530

    net/openvswitch: Use with RCU_INIT_POINTER(x, NULL) in vport-gre.c

    This patch replaces rcu_assign_pointer(x, NULL) with RCU_INIT_POINTER(x, NULL)

    The rcu_assign_pointer() ensures that the initialization of a structure
    is carried out before storing a pointer to that structure.
    And in the case of the NULL pointer, there is no structure to initialize.
    So, rcu_assign_pointer(p, NULL) can be safely converted to RCU_INIT_POINTER(p, NULL)

    Signed-off-by: Monam Agarwal <monamagarwal123@gmail.com>
    Signed-off-by: Jesse Gross <jesse@nicira.com>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/vport-gre.c b/net/openvswitch/vport-gre.c
index bcfcbaa..7c8c07f 100644
--- a/net/openvswitch/vport-gre.c
+++ b/net/openvswitch/vport-gre.c
@@ -257,7 +257,7 @@ static void gre_tnl_destroy(struct vport *vport)
 
  ovs_net = net_generic(net, ovs_net_id);
 
- rcu_assign_pointer(ovs_net->vport_net.gre_vport, NULL);
+ RCU_INIT_POINTER(ovs_net->vport_net.gre_vport, NULL);
  ovs_vport_deferred_free(vport);
  gre_exit();
 }
-- 
1.7.1